### PR TITLE
Weather's effect on fire is lessened + Weather now effects acid >:)

### PR DESCRIPTION
**Pull request by kugamo**
# About the pull request

The numbers behind different weather events smothering fire has been nerfed across the board. While still strong, blizzards for example shouldn't INSTANTLY put out fires (this is the strongest weather on soro). Its more subtle now.

Xenos' acid is effected by weather events too now: damaging less, lasting shorter, and with a very small chance to disappear on cades.

# Explain why it's good for the game

Expands weather effects disadvantages to actions of xenos when before it only hurt marines. B A L A N C E :0)

Muh environmental realism.

# Changelog

:cl:
add: Weather now effects acid, lasting shorter and damaging less.
balance: Weather's effects on fires have been nerfed by a few points.
/:cl:

### CODE REVIEW

File: code/datums/effects/acid.dm [hunk 1]
```diff
@@ -92,3 +96,20 @@
...
+/datum/effects/acid/proc/handle_weather()
+	SIGNAL_HANDLER
+
+	var/area/A = get_area(src)
+	if(!A)
+		return
+
+	if(SSweather.is_weather_event && locate(A.master) in SSweather.weather_areas)
+		//smothering_strength is 1-10, we use this to take a proportional amount off the stats
+		duration = duration - (duration * (SSweather.weather_event_instance.fire_smothering_strength / 10))
```
Comment (carlarctg) [inline code/datums/effects/acid.dm:109]: you really should make these multipliers, into x * 0.1, it's usually not a concern but for procs that are ran more than once in a blue moon it's good practice

File: code/game/objects/effects/aliens.dm [hunk 2]
```diff
@@ -311,12 +312,29 @@
...
 
+/obj/effect/xenomorph/acid/proc/handle_weather()
+	SIGNAL_HANDLER
+
+	var/area/A = get_area(src)
+	if(!A)
+		return
+
+	if(SSweather.is_weather_event && locate(A.master) in SSweather.weather_areas)
+		acid_strength = acid_strength + (SSweather.weather_event_instance.fire_smothering_strength / 3) //smothering_strength is 1-10, acid strench is a multiplier
```
Comment (carlarctg) [inline code/game/objects/effects/aliens.dm:332]: ditto

File: code/game/objects/effects/aliens.dm [hunk 3]
```diff
@@ -311,12 +312,29 @@
...
+	RegisterSignal(SSdcs, COMSIG_GLOB_WEATHER_CHANGE, .proc/handle_weather)
+
 /obj/effect/xenomorph/acid/Destroy()
 	acid_t = null
 	. = ..()
 
+/obj/effect/xenomorph/acid/proc/handle_weather()
+	SIGNAL_HANDLER
+
+	var/area/A = get_area(src)
```
Comment (carlarctg) [inline code/game/objects/effects/aliens.dm:327]: don't use single letter variables

File: code/datums/effects/acid.dm [hunk 4]
```diff
@@ -92,3 +96,20 @@
 	if(acid_goopiness <= 0)
 		return TRUE
 	else return FALSE
+
+/datum/effects/acid/proc/handle_weather()
+	SIGNAL_HANDLER
+
+	var/area/A = get_area(src)
```
Comment (carlarctg) [inline code/datums/effects/acid.dm:103]: single leter

**Review by harryob — CHANGES_REQUESTED** [review]
per mr arc

**Review by harryob — APPROVED** [review]

**Pull request merged**